cirq/contrib/noise_models/noise_models.py

```diff
@@ -52,93 +54,134 @@ def __init__(self, depol_prob: float):
+    This simulates asymmetric readout error. Note that since noise is applied
+    before the measurement moment, composing this model on top of another noise
+    model will place the T1 decay immediately before the measurement
+    (regardless of the previously-added noise).
 
     If a circuit contains measurements, they must be in moments that don't
     also contain gates.
     """
 
-    def __init__(
-            self,
-            depol_prob: float,
-            bitflip_prob: float,
-            decay_prob: float,
-    ):
+    def __init__(self, decay_prob: float):
         """A depolarizing noise model with damped readout error.
 
         Args:
-            depol_prob: Depolarizing probability.
-            bitflip_prob: Probability of a bit-flip during measurement.
             decay_prob: Probability of T1 decay during measurement.
-                Bitflip noise is applied first, then amplitude decay.
         """
-        value.validate_probability(depol_prob, 'depol prob')
-        value.validate_probability(bitflip_prob, 'bitflip prob')
         value.validate_probability(decay_prob, 'decay_prob')
-        self.qubit_noise_gate = ops.DepolarizingChannel(depol_prob)
-        self.readout_noise_gate = ops.BitFlipChannel(bitflip_prob)
         self.readout_decay_gate = ops.AmplitudeDampingChannel(decay_prob)
 
     def noisy_moment(self, moment: 'cirq.Moment',
                      system_qubits: Sequence['cirq.Qid']):
+        if self.is_virtual_moment(moment):
+            return moment
         if _homogeneous_moment_is_measurements(moment):
             return [
-                ops.Moment(self.readout_decay_gate(q) for q in system_qubits),
-                ops.Moment(self.readout_noise_gate(q) for q in system_qubits),
-                moment
+                ops.Moment(
+                    self.readout_decay_gate(q).with_tags(ops.VirtualTag())
+                    for q in system_qubits), moment
             ]
+        return moment
+
+
+class PerQubitDepolarizingNoiseModel(devices.NoiseModel):
+    """DepolarizingNoiseModel which allows depolarization probabilities to be
+    specified separately for each qubit.
+
+    Similar to depol_prob in DepolarizingNoiseModel, depol_prob_map should map
+    Qids in the device to their depolarization probability.
+    """
+
+    def __init__(
+            self,
+            depol_prob_map: Dict['cirq.Qid', float],
+    ):
+        """A depolarizing noise model with variable per-qubit noise.
+
+        Args:
+            depol_prob_map: Map of depolarizing probabilities for each qubit.
+        """
+        for qubit, depol_prob in depol_prob_map.items():
+            value.validate_probability(depol_prob, f'depol prob of {qubit}')
+        self.depol_prob_map = depol_prob_map
+
+    def noisy_moment(self, moment: 'cirq.Moment',
+                     system_qubits: Sequence['cirq.Qid']):
+        if _homogeneous_moment_is_measurements(moment):
```

Comment:
Does this need a is_virtual_moment check?

Reply:
Good point - done.